bitcoin/exchange/bitcoinprices.go

```diff
@@ -125,66 +143,31 @@ func (b *BitcoinAverage) decode(body io.ReadCloser) (err error) {
 			if !ok {
 				return errors.New("Type assertion failed")
 			}
-			b.cache[k] = price
+			cache[k] = price
 		}
 	}
 	return nil
 }
 
-type BitPay struct {
-	cache  map[string]float64
-	client *http.Client
-}
-
-func (b *BitPay) fetch() (err error) {
-	resp, err := b.client.Get("https://bitpay.com/api/rates")
-	if err != nil {
-		return err
-	}
-	return b.decode(resp.Body)
-}
-
-func (b *BitPay) decode(body io.ReadCloser) (err error) {
-	decoder := json.NewDecoder(body)
-	var data []map[string]interface{}
-	err = decoder.Decode(&data)
-	if err != nil {
-		return err
-	}
+func (b BitPayDecoder) decode(dat interface{}, cache map[string]float64) (err error) {
+	data := dat.([]interface{})
```

Comment:
BitPay's peculiar `[{},{},...]` JSON output, needs a different cast than the rest.